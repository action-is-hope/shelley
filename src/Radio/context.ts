import type { RadioGroupState } from "react-stately";
import { useContext, createContext } from "react";
import type { Validation } from "../typings/shared-types";
interface RadioGroupContextValue {
  name?: string;
  isInvalid?: Validation["isInvalid"];
  state: RadioGroupState;
}

export const RadioGroupContext =
  createContext<RadioGroupContextValue | null>(null);

export function useRadioGroupProvider() {
  const context = useContext(RadioGroupContext);
  // If context is null, Radio used outside of our provider so throw an error.
  if (context === null) {
    throw Error(
      `Radio must be used inside a RadioGroup, else it will not function correctly.`
    );
  }
  return context;
}
