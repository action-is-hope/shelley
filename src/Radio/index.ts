export * from "./Radio";
export * from "./RadioGroup";
export * from "./context";
