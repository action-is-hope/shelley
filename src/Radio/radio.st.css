@st-namespace "Radio";
@st-import InputLabel from "../Label/label.st.css";

.root {
  -st-states:
    isDisabled,
    isFocusVisible,
    validationState(enum(valid, invalid)),
    size(number);
  -st-extends: InputLabel;
}

.inputContainer {}

.input {}
