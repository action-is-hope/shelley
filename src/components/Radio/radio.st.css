@namespace "Radio";

@st-import InputLabel from "../../components/Label/label.st.css";

.root {
  -st-states:
    isDisabled,
    isFocusVisible,
    validationState(enum(valid, invalid)),
    vol(number);
}

.inputLabel {
  -st-extends: InputLabel;
}

.inputContainer {}

.input {}
