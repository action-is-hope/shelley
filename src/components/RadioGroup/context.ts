import type { RadioGroupState } from "react-stately";
import { useContext, createContext } from "react";

interface RadioGroupContextValue {
  name?: string;
  validationState?: "valid" | "invalid";
  state: RadioGroupState;
}

export const RadioGroupContext =
  createContext<RadioGroupContextValue | null>(null);

export function useRadioGroupProvider() {
  const context = useContext(RadioGroupContext);
  // If context is null, Radio used outside of our provider so throw an error.
  if (context === null) {
    throw Error(
      `Radio must be used inside a RadioGroup, else it will not function correctly.`
    );
  }
  return context;
}
