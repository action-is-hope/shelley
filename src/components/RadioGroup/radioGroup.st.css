/** radioGroup.st.css */
@namespace "RadioGroup";

:import {
  -st-from: '../Field/field.st.css';
  -st-default: Field
}

.root {
  -st-extends: Field;
  -st-states: orientation(enum(vertical, horizontal));
}

.group {
  display: flex;
}

.root:orientation(vertical) .group {
  flex-direction: column;
}

.root:orientation(horizontal) .group {
  flex-flow: wrap;
}
