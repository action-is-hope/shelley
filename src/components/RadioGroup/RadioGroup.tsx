import { Ref, forwardRef, ReactElement } from "react";
import { Field } from "../Field/Field";
import type { FieldProps } from "../Field/Field";

import { useRadioGroupState } from "react-stately";
import { useRadioGroup } from "react-aria";

import type { AriaRadioGroupProps } from "@react-types/radio";
import type { RadioProps } from "../Radio/Radio";
import { RadioGroupContext } from "./context";
/* = Style API. */
import { st, classes } from "./radioGroup.st.css";
import type { Orientation } from "src/typings/shared-types";

export interface RadioGroupProps
  extends Omit<AriaRadioGroupProps, "excludeFromTabOrder">,
    Omit<
      FieldProps,
      | "label"
      | "startAdornment"
      | "endAdornment"
      | "disableLabelTransition"
      | "hasValue"
      | "variant"
    > {
  /**
   * Position of the label.
   * @default "over"
   */
  labelPosition?: "top" | "side" | "hidden";
  /** Class */
  className?: string;
  /**
   * The Radios contained within the RadioGroup.
   */
  children: ReactElement<RadioProps> | ReactElement<RadioProps>[];
  /**
   * The axis the radios should align with.
   * @default 'vertical'
   */
  orientation?: Orientation;
}

function RadioGroup(props: RadioGroupProps, ref?: Ref<HTMLDivElement>) {
  const {
    className: classNameProp,
    description,
    isDisabled,
    errorMessage,
    validationState,
    label,
    labelPosition = "top",
    orientation = "vertical",
    vol,
    children,
    "data-id": dataId,
  } = props;

  const state = useRadioGroupState(props);
  const { labelProps, radioGroupProps, descriptionProps, errorMessageProps } =
    useRadioGroup(props, state);

  return (
    <Field
      {...{
        disabled: isDisabled,
        errorMessage,
        validationState,
        errorMessageProps,
        fieldContainerProps: { ...radioGroupProps, className: classes.group },
        description,
        descriptionProps,
        label,
        labelPosition,
        labelProps,
        vol,
        variant: false,
        ref,
        "data-id": dataId,
      }}
      className={st(classes.root, { orientation }, classNameProp)}
    >
      <RadioGroupContext.Provider
        value={{
          validationState,
          state,
        }}
      >
        {children}
      </RadioGroupContext.Provider>
    </Field>
  );
}

const _RadioGroup = forwardRef(RadioGroup);
export { _RadioGroup as RadioGroup };
