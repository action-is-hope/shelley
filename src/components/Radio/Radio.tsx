import React, { forwardRef, useRef } from "react";
import type { AriaRadioProps } from "@react-types/radio";
import { useRadio } from "react-aria";
import { mergeRefs, mergeProps } from "@react-aria/utils";
import { useFocusRing } from "react-aria";
import type { Volume, AlignPos, ComponentBase } from "../types";
import { useRadioGroupProvider } from "../RadioGroup/context";
import Label from "../Label/Label";
/* = Style API. */
import { st, classes } from "./radio.st.css";

export interface RadioProps extends AriaRadioProps, ComponentBase {
  className?: string;
  /** The position of the label relative to the label. */
  inputPosition?: AlignPos;
  /** Visually hide the label so it is still accessible to assistive technologies. */
  visuallyHideLabel?: boolean;
  /** How 'loud' should this input be? */
  vol?: Volume;
}

const Radio = forwardRef(
  (props: RadioProps, ref?: React.Ref<HTMLInputElement>) => {
    const {
      className: classNameProp,
      children,
      // validationState,
      visuallyHideLabel,
      inputPosition,
      vol = 1,
      isDisabled,
      "data-id": dataId,
    } = props;
    // const localRef = useRef(null);
    const inputRef = useRef<HTMLInputElement>(null);
    // const groupState = useContext(RadioGroupContext);
    // const { inputProps } = useRadio(props, groupState, ref);
    const radioGroupProps = useRadioGroupProvider();
    const { validationState, state } = radioGroupProps;

    const { inputProps } = useRadio(
      {
        ...props,
        ...radioGroupProps,
        isDisabled,
      },
      state,
      inputRef
    );

    const { isFocusVisible, focusProps } = useFocusRing();

    const classNames = st(
      classes.root,
      {
        isDisabled,
        isFocusVisible,
        validationState,
        vol: vol ? vol : undefined,
      },
      classNameProp
    );

    const inputControl = (
      <span className={classes.inputContainer}>
        <input
          className={classes.input}
          {...mergeProps(inputProps, focusProps)}
          ref={ref ? mergeRefs(ref, inputRef) : inputRef}
          data-id={dataId ? `${dataId}--input` : undefined}
        />
      </span>
    );

    return (
      <>
        {children ? (
          <Label
            className={classNames}
            {...{ inputControl, inputPosition }}
            visuallyHidden={visuallyHideLabel}
            data-id={dataId ? `${dataId}--label` : undefined}
          >
            {children}
          </Label>
        ) : (
          <span
            className={classNames}
            data-id={dataId ? `${dataId}--noLabel` : undefined}
          >
            {inputControl}
          </span>
        )}
      </>
    );
  }
);

export default Radio;

Radio.displayName = "Radio";
