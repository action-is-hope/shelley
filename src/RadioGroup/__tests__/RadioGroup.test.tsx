import { Radio } from "../../Radio/Radio";
import { RadioGroup } from "../RadioGroup";
import renderer from "react-test-renderer";
// useID breaks snapshots so using the SSrProvider.
import { SSRProvider } from "react-aria";

describe("Radio", () => {
  it("renders correctly with label", () => {
    const tree = renderer
      .create(
        <SSRProvider>
          <RadioGroup label="Radio Group">
            <Radio value="test1">One</Radio>
            <Radio value="test2">Two</Radio>
          </RadioGroup>
        </SSRProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders as vertical", () => {
    const tree = renderer
      .create(
        <SSRProvider>
          <RadioGroup label="Radio Group" orientation="vertical">
            <Radio value="test1">One</Radio>
            <Radio value="test2">Two</Radio>
          </RadioGroup>
        </SSRProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders with size '1'", () => {
    const tree = renderer
      .create(
        <SSRProvider>
          <RadioGroup label="Radio Group">
            <Radio value="test1" size={1}>
              One
            </Radio>
            <Radio value="test2" size={1}>
              Two
            </Radio>
          </RadioGroup>
        </SSRProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Renders with size '2'", () => {
    const tree = renderer
      .create(
        <SSRProvider>
          <RadioGroup label="Radio Group">
            <Radio value="test1" size={2}>
              One
            </Radio>
            <Radio value="test2" size={2}>
              Two
            </Radio>
          </RadioGroup>
        </SSRProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Renders with size '3'", () => {
    const tree = renderer
      .create(
        <SSRProvider>
          <RadioGroup label="Radio Group">
            <Radio value="test1" size={3}>
              One
            </Radio>
            <Radio value="test2" size={3}>
              Two
            </Radio>
          </RadioGroup>
        </SSRProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Renders with size '4'", () => {
    const tree = renderer
      .create(
        <SSRProvider>
          <RadioGroup label="Radio Group">
            <Radio value="test1" size={4}>
              One
            </Radio>
            <Radio value="test2" size={4}>
              Two
            </Radio>
          </RadioGroup>
        </SSRProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Renders with size '5'", () => {
    const tree = renderer
      .create(
        <SSRProvider>
          <RadioGroup label="Radio Group">
            <Radio value="test1" size={5}>
              One
            </Radio>
            <Radio value="test2" size={5}>
              Two
            </Radio>
          </RadioGroup>
        </SSRProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Renders with size '6'", () => {
    const tree = renderer
      .create(
        <SSRProvider>
          <RadioGroup label="Radio Group">
            <Radio value="test1" size={6}>
              One
            </Radio>
            <Radio value="test2" size={6}>
              Two
            </Radio>
          </RadioGroup>
        </SSRProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
