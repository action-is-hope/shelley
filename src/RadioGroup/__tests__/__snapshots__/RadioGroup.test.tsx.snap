// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Renders with size '1' 1`] = `
<div
  className="Field__Test__root Field__Test---vol-1-1 Field__Test---labelPosition-3-top RadioGroup__Test__root RadioGroup__Test---orientation-8-vertical"
>
  <label
    className="Label__Test__root Field__Test__inputLabel"
    id="react-aria-2"
  >
    <span
      className="Label__Test__text"
    >
      Radio Group
    </span>
  </label>
  <div
    aria-labelledby="react-aria-2"
    aria-orientation="vertical"
    className="Field__Test__fieldContainer RadioGroup__Test__group"
    id="react-aria-1"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="radiogroup"
  >
    <label
      className="Label__Test__root Label__Test--hasInput Radio__Test__root Radio__Test---size-1-1"
    >
      <span
        className="Label__Test__text"
      >
        One
      </span>
      <span
        className="Radio__Test__inputContainer"
      >
        <input
          checked={false}
          className="Radio__Test__input"
          disabled={false}
          name="react-aria-5"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          required={false}
          tabIndex={0}
          type="radio"
          value="test1"
        />
      </span>
    </label>
    <label
      className="Label__Test__root Label__Test--hasInput Radio__Test__root Radio__Test---size-1-1"
    >
      <span
        className="Label__Test__text"
      >
        Two
      </span>
      <span
        className="Radio__Test__inputContainer"
      >
        <input
          checked={false}
          className="Radio__Test__input"
          disabled={false}
          name="react-aria-5"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          required={false}
          tabIndex={0}
          type="radio"
          value="test2"
        />
      </span>
    </label>
  </div>
</div>
`;

exports[`Radio Renders with size '2' 1`] = `
<div
  className="Field__Test__root Field__Test---vol-1-1 Field__Test---labelPosition-3-top RadioGroup__Test__root RadioGroup__Test---orientation-8-vertical"
>
  <label
    className="Label__Test__root Field__Test__inputLabel"
    id="react-aria-2"
  >
    <span
      className="Label__Test__text"
    >
      Radio Group
    </span>
  </label>
  <div
    aria-labelledby="react-aria-2"
    aria-orientation="vertical"
    className="Field__Test__fieldContainer RadioGroup__Test__group"
    id="react-aria-1"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="radiogroup"
  >
    <label
      className="Label__Test__root Label__Test--hasInput Radio__Test__root Radio__Test---size-1-2"
    >
      <span
        className="Label__Test__text"
      >
        One
      </span>
      <span
        className="Radio__Test__inputContainer"
      >
        <input
          checked={false}
          className="Radio__Test__input"
          disabled={false}
          name="react-aria-5"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          required={false}
          tabIndex={0}
          type="radio"
          value="test1"
        />
      </span>
    </label>
    <label
      className="Label__Test__root Label__Test--hasInput Radio__Test__root Radio__Test---size-1-2"
    >
      <span
        className="Label__Test__text"
      >
        Two
      </span>
      <span
        className="Radio__Test__inputContainer"
      >
        <input
          checked={false}
          className="Radio__Test__input"
          disabled={false}
          name="react-aria-5"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          required={false}
          tabIndex={0}
          type="radio"
          value="test2"
        />
      </span>
    </label>
  </div>
</div>
`;

exports[`Radio Renders with size '3' 1`] = `
<div
  className="Field__Test__root Field__Test---vol-1-1 Field__Test---labelPosition-3-top RadioGroup__Test__root RadioGroup__Test---orientation-8-vertical"
>
  <label
    className="Label__Test__root Field__Test__inputLabel"
    id="react-aria-2"
  >
    <span
      className="Label__Test__text"
    >
      Radio Group
    </span>
  </label>
  <div
    aria-labelledby="react-aria-2"
    aria-orientation="vertical"
    className="Field__Test__fieldContainer RadioGroup__Test__group"
    id="react-aria-1"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="radiogroup"
  >
    <label
      className="Label__Test__root Label__Test--hasInput Radio__Test__root Radio__Test---size-1-3"
    >
      <span
        className="Label__Test__text"
      >
        One
      </span>
      <span
        className="Radio__Test__inputContainer"
      >
        <input
          checked={false}
          className="Radio__Test__input"
          disabled={false}
          name="react-aria-5"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          required={false}
          tabIndex={0}
          type="radio"
          value="test1"
        />
      </span>
    </label>
    <label
      className="Label__Test__root Label__Test--hasInput Radio__Test__root Radio__Test---size-1-3"
    >
      <span
        className="Label__Test__text"
      >
        Two
      </span>
      <span
        className="Radio__Test__inputContainer"
      >
        <input
          checked={false}
          className="Radio__Test__input"
          disabled={false}
          name="react-aria-5"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          required={false}
          tabIndex={0}
          type="radio"
          value="test2"
        />
      </span>
    </label>
  </div>
</div>
`;

exports[`Radio Renders with size '4' 1`] = `
<div
  className="Field__Test__root Field__Test---vol-1-1 Field__Test---labelPosition-3-top RadioGroup__Test__root RadioGroup__Test---orientation-8-vertical"
>
  <label
    className="Label__Test__root Field__Test__inputLabel"
    id="react-aria-2"
  >
    <span
      className="Label__Test__text"
    >
      Radio Group
    </span>
  </label>
  <div
    aria-labelledby="react-aria-2"
    aria-orientation="vertical"
    className="Field__Test__fieldContainer RadioGroup__Test__group"
    id="react-aria-1"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="radiogroup"
  >
    <label
      className="Label__Test__root Label__Test--hasInput Radio__Test__root Radio__Test---size-1-4"
    >
      <span
        className="Label__Test__text"
      >
        One
      </span>
      <span
        className="Radio__Test__inputContainer"
      >
        <input
          checked={false}
          className="Radio__Test__input"
          disabled={false}
          name="react-aria-5"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          required={false}
          tabIndex={0}
          type="radio"
          value="test1"
        />
      </span>
    </label>
    <label
      className="Label__Test__root Label__Test--hasInput Radio__Test__root Radio__Test---size-1-4"
    >
      <span
        className="Label__Test__text"
      >
        Two
      </span>
      <span
        className="Radio__Test__inputContainer"
      >
        <input
          checked={false}
          className="Radio__Test__input"
          disabled={false}
          name="react-aria-5"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          required={false}
          tabIndex={0}
          type="radio"
          value="test2"
        />
      </span>
    </label>
  </div>
</div>
`;

exports[`Radio Renders with size '5' 1`] = `
<div
  className="Field__Test__root Field__Test---vol-1-1 Field__Test---labelPosition-3-top RadioGroup__Test__root RadioGroup__Test---orientation-8-vertical"
>
  <label
    className="Label__Test__root Field__Test__inputLabel"
    id="react-aria-2"
  >
    <span
      className="Label__Test__text"
    >
      Radio Group
    </span>
  </label>
  <div
    aria-labelledby="react-aria-2"
    aria-orientation="vertical"
    className="Field__Test__fieldContainer RadioGroup__Test__group"
    id="react-aria-1"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="radiogroup"
  >
    <label
      className="Label__Test__root Label__Test--hasInput Radio__Test__root Radio__Test---size-1-5"
    >
      <span
        className="Label__Test__text"
      >
        One
      </span>
      <span
        className="Radio__Test__inputContainer"
      >
        <input
          checked={false}
          className="Radio__Test__input"
          disabled={false}
          name="react-aria-5"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          required={false}
          tabIndex={0}
          type="radio"
          value="test1"
        />
      </span>
    </label>
    <label
      className="Label__Test__root Label__Test--hasInput Radio__Test__root Radio__Test---size-1-5"
    >
      <span
        className="Label__Test__text"
      >
        Two
      </span>
      <span
        className="Radio__Test__inputContainer"
      >
        <input
          checked={false}
          className="Radio__Test__input"
          disabled={false}
          name="react-aria-5"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          required={false}
          tabIndex={0}
          type="radio"
          value="test2"
        />
      </span>
    </label>
  </div>
</div>
`;

exports[`Radio Renders with size '6' 1`] = `
<div
  className="Field__Test__root Field__Test---vol-1-1 Field__Test---labelPosition-3-top RadioGroup__Test__root RadioGroup__Test---orientation-8-vertical"
>
  <label
    className="Label__Test__root Field__Test__inputLabel"
    id="react-aria-2"
  >
    <span
      className="Label__Test__text"
    >
      Radio Group
    </span>
  </label>
  <div
    aria-labelledby="react-aria-2"
    aria-orientation="vertical"
    className="Field__Test__fieldContainer RadioGroup__Test__group"
    id="react-aria-1"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="radiogroup"
  >
    <label
      className="Label__Test__root Label__Test--hasInput Radio__Test__root Radio__Test---size-1-6"
    >
      <span
        className="Label__Test__text"
      >
        One
      </span>
      <span
        className="Radio__Test__inputContainer"
      >
        <input
          checked={false}
          className="Radio__Test__input"
          disabled={false}
          name="react-aria-5"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          required={false}
          tabIndex={0}
          type="radio"
          value="test1"
        />
      </span>
    </label>
    <label
      className="Label__Test__root Label__Test--hasInput Radio__Test__root Radio__Test---size-1-6"
    >
      <span
        className="Label__Test__text"
      >
        Two
      </span>
      <span
        className="Radio__Test__inputContainer"
      >
        <input
          checked={false}
          className="Radio__Test__input"
          disabled={false}
          name="react-aria-5"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          required={false}
          tabIndex={0}
          type="radio"
          value="test2"
        />
      </span>
    </label>
  </div>
</div>
`;

exports[`Radio renders as vertical 1`] = `
<div
  className="Field__Test__root Field__Test---vol-1-1 Field__Test---labelPosition-3-top RadioGroup__Test__root RadioGroup__Test---orientation-8-vertical"
>
  <label
    className="Label__Test__root Field__Test__inputLabel"
    id="react-aria-2"
  >
    <span
      className="Label__Test__text"
    >
      Radio Group
    </span>
  </label>
  <div
    aria-labelledby="react-aria-2"
    aria-orientation="vertical"
    className="Field__Test__fieldContainer RadioGroup__Test__group"
    id="react-aria-1"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="radiogroup"
  >
    <label
      className="Label__Test__root Label__Test--hasInput Radio__Test__root Radio__Test---size-1-1"
    >
      <span
        className="Label__Test__text"
      >
        One
      </span>
      <span
        className="Radio__Test__inputContainer"
      >
        <input
          checked={false}
          className="Radio__Test__input"
          disabled={false}
          name="react-aria-5"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          required={false}
          tabIndex={0}
          type="radio"
          value="test1"
        />
      </span>
    </label>
    <label
      className="Label__Test__root Label__Test--hasInput Radio__Test__root Radio__Test---size-1-1"
    >
      <span
        className="Label__Test__text"
      >
        Two
      </span>
      <span
        className="Radio__Test__inputContainer"
      >
        <input
          checked={false}
          className="Radio__Test__input"
          disabled={false}
          name="react-aria-5"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          required={false}
          tabIndex={0}
          type="radio"
          value="test2"
        />
      </span>
    </label>
  </div>
</div>
`;

exports[`Radio renders correctly with label 1`] = `
<div
  className="Field__Test__root Field__Test---vol-1-1 Field__Test---labelPosition-3-top RadioGroup__Test__root RadioGroup__Test---orientation-8-vertical"
>
  <label
    className="Label__Test__root Field__Test__inputLabel"
    id="react-aria-2"
  >
    <span
      className="Label__Test__text"
    >
      Radio Group
    </span>
  </label>
  <div
    aria-labelledby="react-aria-2"
    aria-orientation="vertical"
    className="Field__Test__fieldContainer RadioGroup__Test__group"
    id="react-aria-1"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="radiogroup"
  >
    <label
      className="Label__Test__root Label__Test--hasInput Radio__Test__root Radio__Test---size-1-1"
    >
      <span
        className="Label__Test__text"
      >
        One
      </span>
      <span
        className="Radio__Test__inputContainer"
      >
        <input
          checked={false}
          className="Radio__Test__input"
          disabled={false}
          name="react-aria-5"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          required={false}
          tabIndex={0}
          type="radio"
          value="test1"
        />
      </span>
    </label>
    <label
      className="Label__Test__root Label__Test--hasInput Radio__Test__root Radio__Test---size-1-1"
    >
      <span
        className="Label__Test__text"
      >
        Two
      </span>
      <span
        className="Radio__Test__inputContainer"
      >
        <input
          checked={false}
          className="Radio__Test__input"
          disabled={false}
          name="react-aria-5"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragStart={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          required={false}
          tabIndex={0}
          type="radio"
          value="test2"
        />
      </span>
    </label>
  </div>
</div>
`;
